### ISSUE TITLE
Update button not enabled after deleting a coordinator's fee

### ISSUE THREAD

**Issue by NickWeir63** [openfoodfoundation/openfoodnetwork#1165]
When i delete a coordinators fee from an order cycle the update button stays greyed out.   I can only save this change if i make  another change to the order cycle which then enables the update button
![screenshot 2016-09-22 13 07 20](https://cloud.githubusercontent.com/assets/10517702/18748156/e5f36134-80c8-11e6-84ec-38a94a89d19a.png)

**Comment by stveep** [comment]
This could be something for someone to get started on, if you like a bit of Angular. I think it's likely to be a function in `app/assets/javascripts/admin/order_cycles/services/order_cycle.js.coffee` not correctly flagging that the form has unsaved changes ('dirty').

**Comment by lin-d-hop** [comment]
@OliverUK Would you be willing to test this? If so let me know when and I'll get it on UK staging for you.

**Comment by OliverUK** [comment]
Happy to do some user testing @lin-d-hop this Sunday or Mondya.

**Comment by OliverUK** [comment]
1) Logged in as manager of Stroudco enterprise on staging using Firefox 50.0
2) Created a new order cycle and added three suppliers.
![screen shot 2016-11-28 at 12 01 46](https://cloud.githubusercontent.com/assets/17590921/20667747/e2e15d88-b562-11e6-8b4c-b308b48bffbd.png)
3) Added a coordinator fee and save buttons appear as expected.
![screen shot 2016-11-28 at 12 02 17](https://cloud.githubusercontent.com/assets/17590921/20667770/0da4760e-b563-11e6-949b-957420ad3a8b.png)
4) Removed coordinators fee and save button now also appears.
![screen shot 2016-11-28 at 12 02 35](https://cloud.githubusercontent.com/assets/17590921/20667796/25c13056-b563-11e6-9d70-91baaa471bf7.png)
5) Saved, left screen and returned to find changes saved as expected. 
6) Checked shopfront and prices increase as expected with the addition of a coordinator fee.
7) Created a new enterprise fee for Stroudco and repeated the above with the new fee. Behaved as expected.

**Comment by lin-d-hop** [comment]
Thanks @OliverUK! Great testing description.
@sstead are you happy with Olivers testing above? Any other questions?
If you are happy I'll move this over to milestone v1.10 for release...whoop!

**Comment by sstead** [comment]
Great work @OliverUK. I would also check that this works for a Producer Shop order cycle (simple OC), and check that fees applied to producers and the distributors can also be removed.

**Comment by OliverUK** [comment]
OK I've set myself up as a producer but am unable to create an order cycle for Oliver Muller. I'm logged in using the same email address that is a manager for Stroudco and Oliver Muller but it doesn't offer me to create an order cycle for Oliver Muller. It also tells me that email confirmation is pending for my email address in case that is related? So haven't been able to test on Producer Shop OC @sstead  @lin-d-hop

**Comment by lin-d-hop** [comment]
@oliveruk you might need to ensure your enterprise is a shopfront enterprise and not a profile only enterprise

**Comment by OliverUK** [comment]
![screen shot 2016-11-30 at 12 11 25](https://cloud.githubusercontent.com/assets/17590921/20751871/298d2f02-b6f6-11e6-8893-de14a1f4d3a4.png)

**Comment by NickWeir63** [comment]
i have increased enterprise limit on @OliverUK account to 100 in case this was the problem.  If this doesn't work Oliver you can log in to may account on staging nick@nickweir.co.uk password localfood1

**Comment by OliverUK** [comment]
@lin-d-hop @NickWeir63 
I've still not been able to test this on a producer's order cycle and I suspect it's due to us not being able to confirm email addresses on staging. 
For my enterprise "Oliver Muller" my email address is pending confirmation. I made Nick's email address a manager, logged out and logged in as Nick and still can't create a new OC for anything other than Stroudco. Went back in as me and tried to make Nick's email the one for notifications but if I then save and go back to it I find it looks the same as before, i.e. still waiting for email confirmation from me. 

I suspect OFN is also wanting to have Nick's email address confirmed. I may be wrong but I think if you have one email address pending and then try another one which is also not known to OFN, the screen will still show that it's waiting for the first of the two. 
I also tried spree@example.com but with the same result.

**Comment by NickWeir63** [comment]
@OliverUK  i will phone you with the superadmin login so that you can create a new enterprise that way

**Comment by OliverUK** [comment]
Thanks Nick. Yes, I was able to create an OC for a confirmed user (I used Nick's Test Producer).
I added a coordinator fee and saved. The front end result was as expected. I was able to delete the fee again from the OC and save. The front end once again responded. 

I tried as per @sstead the same with fees applied on the incoming supplier and outgoing distributor and was able to add and remove them and save.
